client/src/templates/Challenges/utils/frame.ts

```diff
@@ -267,9 +267,44 @@ const writeContentToFrame = (frameContext: Context) => {
     createHeader(frameContext.element.id) + frameContext.build,
     frameContext.document
   );
+
+  registerScrollEventListner(frameContext.element);
+
+  if (getPreviewScrollPosition()) {
+    restorePreviewScrollPosition(
+      frameContext.element,
+      getPreviewScrollPosition()
+    );
+  }
   return frameContext;
 };
 
+let previewScrollPosition = 0;
+export const getPreviewScrollPosition = () => previewScrollPosition;
+export const setPreviewScrollPosition = (position: number) =>
+  (previewScrollPosition = position);
+
+const registerScrollEventListner = (iframe: HTMLIFrameElement) => {
+  iframe.contentDocument?.addEventListener(
+    'scroll',
+    (event: IframeEvent<Document>) => {
+      const { target } = event;
+      if (target.body.scrollTop) {
+        setPreviewScrollPosition(target.body.scrollTop);
```

Comment:
```suggestion
      const currentTarget = event.currentTarget as Document | null;
      if (currentTarget?.body.scrollTop) {
        setPreviewScrollPosition(currentTarget?.body.scrollTop);
      }
```
`currentTarget` is safer because we know that it's `iframe.contentDocument` because that's what the listener was attached to.